posts/inside-rust/2021-05-15-What-the-error-handling-project-group-is-working-towards.md

```diff
@@ -0,0 +1,235 @@
+---
+layout: post
+title: "What the Error Handling Project Group is Working Towards"
+author: Jane Lusby
+team: the library team <https://www.rust-lang.org/governance/teams/library>
+---
+
+This blog post is a follow up of our [previous](https://blog.rust-lang.org/inside-rust/2020/11/23/What-the-error-handling-project-group-is-working-on.html) post detailing what we're working on now. We've been iterating for a while now on some of the challenges that we see with error handling today and have reached the point where we want to describe some of the new changes we're working towards. But first we need to describe the main challenges we've identified.
+
+> Disclaimer: *This post is equal parts plan and aspiration. There are technical challenges here to sort out so the final outcome may look rather different from our initial vision, so please don't assume any of this is final.*
+
+## Error Handling Today
+
+The first challenge we'd like to solve is that it's easy to lose context accidentally when reporting errors. There are a couple of places this can happen, either when printing an error and forgetting to print sources, when returning an error from main, or when converting a recoverable error into a non recoverable error.
+
+Consider this example:
+
+```rust
+use std::fmt;
+
+// We have a program that loads a config and expects that
+// loading the config will always succeed.
+fn main() {
+    let _config = load_config()
+        .expect("config is always valid and exists");
+}
+
+// We have a dummy implementation of load_config which
+// always errors, since we're just focusing on diagnostics
```

Comment:
errors or errs 😄 ?

Reply:
I think errors is appropriate here, pretty sure it's valid as a verb.